include/miopen/miopen.h

```diff
@@ -4704,6 +4706,231 @@ miopenReduceTensor(miopenHandle_t handle,
+
+/*! @enum miopenProblemDirection_t
+ * Directions of miopen operation.
+ */
+typedef enum
+{
+    miopenProblemDirectionForward         = 0,
+    miopenProblemDirectionBackward        = 1,
+    miopenProblemDirectionBackwardWeights = 2,
+} miopenProblemDirection_t;
+
+/*! @enum miopenTensorArgumentId_t
+ * Identifiers for tensor arguments of problems and operations.
+ */
+typedef enum
+{
+    miopenTensorArgumentIdInvalid = 0,
+    miopenTensorConvolutionX      = 1,
+    miopenTensorConvolutionW      = 2,
+    miopenTensorConvolutionY      = 3,
+} miopenTensorArgumentId_t;
+
+/*! @enum miopenTensorArgumentId_t
+ * Different ways to sort results of the find call.
+ */
+typedef enum
+{
+    miopenFindResultsOrderByTime   = 0,
+    miopenFindResultsOrderByMemory = 1,
+} miopenFindResultsOrder_t;
+
+/*! @brief Initializes a problem object describing a convolution operation.
+ *
+ * @param problem      Pointer to the problem to initialize
+ * @param operatorDesc Descriptor of the operator to be used
+ * @param direction    Direction of the operation
+ * @return             miopenStatus_t
+ */
+MIOPEN_EXPORT miopenStatus_t miopenCreateConvProblem(miopenProblem_t* problem,
+                                                     miopenConvolutionDescriptor_t operatorDesc,
+                                                     miopenProblemDirection_t direction);
+
+/*! @brief Destroys a problem object.
+ *
+ * @param problem Problem to destroy
+ * @return        miopenStatus_t
+ */
+miopenStatus_t miopenDestroyProblem(miopenProblem_t problem);
+
+/*! @brief Sets a tensor descriptor for the specified argument.
+ *
+ * @param problem    Problem to update
+ * @param id         Id of the argument for the descriptor
+ * @param descriptor Tensor descriptor to set
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetProblemTensorDescriptor(miopenProblem_t problem,
+                                                miopenTensorArgumentId_t id,
+                                                const miopenTensorDescriptor_t descriptor);
+
+/*! @brief The miopenFindOptions allows the user to configure how find will be used.
+ */
+MIOPEN_DECLARE_OBJECT(miopenFindOptions);
+
+/*! @brief Initializes miopenFindOptions object.
+ *
+ * @param options    Pointer to options object to initialze
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenCreateFindOptions(miopenFindOptions_t* options);
+
+/*! @brief Destroys miopenFindOptions object.
+ *
+ * @param options    Options object to destroy
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenDestroyFindOptions(miopenFindOptions_t options);
+
+/*! @brief Sets the tuning find option. Default value is zero.
+ *
+ * @param options    Options object to upfate
+ * @param value      Value of zero means no tuning, value of one means tuning enabled
+ * @return           miopenStatus_t
+ */
+miopenStatus_t miopenSetFindOptionTuning(miopenFindOptions_t options, int value);
```

Comment:
It would be better `miopenSetFindOptionEnableTuning(miopenFindOptions_t options, int enabled)`.

Reply:
@DrizztDoUrden @pfultz2 [Recommendation] Use `miopenSetFindOptionTuningMode` and introduce enum for tuning mode (0 - disabled, 1- enabled).